### ISSUE TITLE
journalctl advertises options it doesn't implement

### ISSUE THREAD

**Issue by coretemp** [systemd/systemd#9822]
I was requested to open an issue here via https://github.com/NixOS/nixpkgs/issues/44595#issuecomment-410972640.

## Issue description
From `journalctl --help`:
```
Query the journal.

Options:
  -g --grep=PATTERN          Show entries with MESSSAGE matching PATTERN
     --case-sensitive[=BOOL] Force case sensitive or insenstive matching
```

### Steps to reproduce

```
journalctl --grep a -u sshd.service 
```
returns `Compiled without pattern matching support`

Expected response: 
Lines matching `a`.

## Technical details

- system: `"x86_64-linux"`
 - host os: `Linux 4.14.55, NixOS, 18.03.git.d0c868e (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0.4`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by yuwata** [comment]
It seems that systemd is built without pcre2.

**Comment by yuwata** [comment]
Hmm, we usually not change the contents in help messages or man pages by the compile options...

**Comment by coretemp** [comment]
> Hmm, we usually do not change the contents in help messages or man pages by the compile options...

That's exactly the bug.

**Comment by jtojnar** [comment]
@yuwata There is already a precedent:

https://github.com/systemd/systemd/blob/6f663594bc5ee087cb15175cb691eaebcc45a0a1/src/journal/journalctl.c#L368-L370

**Comment by yuwata** [comment]
In #9826, the options require pcre2 are dropped from only help message.

**Comment by poettering** [comment]
So, as discussed here:

https://github.com/systemd/systemd/pull/9826#issuecomment-411361415

I think we shouldn't alter the `--help` texts based on compile time options. it's a good thing to advertise functionality and have --help and man pages in sync and comprehensive, even if on some systems varying features are turned off.

I have merged #9830 now, that drops the #ifdeffery from the gcrypt bits too in the --help text.
